### PR TITLE
[BUG] Make sure distributor gets passed through to extract_features

### PR DESCRIPTION
**Pull request by marcosfelt**
#### What does this implement/fix? Explain your changes.
TSFresh has a feature that allows users to pass [custom parallelization distributors](https://tsfresh.readthedocs.io/en/latest/text/tsfresh_on_a_cluster.html#parallelization-beyond-a-single-machine). The current implementation of `TSFreshFeatureExtractor` and `TSFreshRelevantFeatureExtractor` do not pass through distributor all the way to `extract_features`. This PR fixes that. 

#### Does your contribution introduce a new dependency? If yes, which one?
N/A

#### Did you add any tests for the change?
~No, but happy to if needed~ Added a MapDistributor test case.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Good spot!

Though we will also need to add this to:

* the `__init__` as an arg and self assign
* to `get_test_params` as an additional test case

Right? Or am I missing sth?

**Comment by marcosfelt** [comment]
> Good spot!
> 
> Though we will also need to add this to:
> 
> * the `__init__` as an arg and self assign
> * to `get_test_params` as an additional test case
> 
> Right? Or am I missing sth?

Thanks for the feedback!

* `distributor` is an arg in `TSFreshFeatureExtractor` (L262) and `TSFreshRelevantFeatureExtractor` (L479). The self assignment happens `_TSFreshFeatureExtractor`, and the assignment to `default_fc_parameters` happens in `_get_extraction_params`.
* I added an additional case that uses the `MapDistributor` as a simple example (no parallelization but should call the relevant methods). I'm not sure if there is a way to assert that the relevant methods were called as they do in the tsfresh tests: https://github.com/blue-yonder/tsfresh/blob/b970a2e5f3f63f264c9ccbd8b0757f01697f3efc/tests/units/feature_extraction/test_extraction.py#L421

File: .all-contributorsrc [hunk 1]
```diff
@@ -3297,6 +3297,12 @@
       ]
     },
     {
+      "login": "marcosfelt",
+      "name": "Kobi Felton",
+      "avatar_url": "https://avatars.githubusercontent.com/u/25933639?v=4",
+      "profile": "https://github.com/marcosfelt",
+      "contributions": [
+        "code"
```
Comment (yarnabrina) [inline .all-contributorsrc:3305]: I think your new additions here are not in right format, and may mess up both yours and Shounak's badges. Can you please fix it, and add as an independent block?

**Review by yarnabrina — CHANGES_REQUESTED** [review]
This is not a review of your main contribution, but just noting something about your modification in the all contributors file.

Also, you may want to run pre-commit locally first, if not done already.

File: .all-contributorsrc (hunk 1 above)
Comment (marcosfelt) [reply .all-contributorsrc:3305]: Fixed! Thanks for noticing this @yarnabrina!

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks for addressing my change requests!

As @yarnabrina says, ensure that the code formatting checks run. Here is the guide on how to set it up:
https://www.sktime.net/en/latest/developer_guide/coding_standards.html

There are code formatting failures in the all-contributors file and the tsfresh file.

**Comment by fkiraly** [comment]
No problem at all - thanks a lot for contributing!

**Review by fkiraly — APPROVED** [review]
Thanks! Made some minor change:

* moved the new parameter to the end to ensure downwards compatibility
* addressed soft dependency isolation in `get_test_params`

**Comment by marcosfelt** [comment]
Thanks @fkiraly! Do I need to do anything for the GH actions checks to run?

**Comment by yarnabrina** [comment]
> Thanks! Made some minor change:
> 
>     * moved the new parameter to the end to ensure downwards compatibility
> 
>     * addressed soft dependency isolation in `get_test_params`

@fkiraly I'd like to object. It's not a new parameter, it was part of `__init__` already. Here's a reference from `main` branch (as of commit a6e12ea524b101cd03a1e1c055e82c33a64f3467), and I think this move will make it a breaking change.

https://github.com/sktime/sktime/blob/main/sktime/transformations/panel/tsfresh.py#L466-L486

**Comment by fkiraly** [comment]
@yarnabrina, I see, thanks for pointing that out! I moved the parameter back to its original position. I was mistaken in assuming that it got added in this PR.

**Review by yarnabrina — APPROVED** [review]

**Pull request merged**